src/gui/addnewtorrentdialog.h

```diff
@@ -65,6 +66,7 @@ class AddNewTorrentDialog: public QDialog
     static bool isTopLevel();
     static void setTopLevel(bool value);
 
+    static void show(QString source, BitTorrent::AddTorrentParams &inParams, QWidget *parent);
```

Comment:
Please no. Unparented dialogue windows are showed on the first screen always in Plasma and maybe in any X session regardless of the main window position.